ophyd/status.py

```diff
@@ -446,6 +468,16 @@ def __init__(self, positioner, target, *, start_ts=None,
         if not self.done:
             self.pos.subscribe(self._notify_watchers,
                                event_type=self.pos.SUB_READBACK)
+        else:
+            self.finish_pos = self.target
+            self.finish_ts = time.time()
+            if self.success:
+                try:
+                    self.pos.est_time.set.record(self)
+                except AttributeError:
+                    print ('est_time.set.record attribute required on {}, but not found'.format(self.pos))
```

Comment:
The problem with printing is that there is no way for the caller -- who might be able to handle this error -- to suppress the print. Instead I suggest this pattern:

```py
except AttributeError:
    raise AttributeError(your_custom_message_here)
```

Reply:
I tried this and kept getting an error, I will try again.